### PR TITLE
Use PyTorch Learning Rate Scheduler

### PR DESCRIPTION
**Pull request by kylegao91**
#37

### CODE REVIEW

File: tests/test_optim_optim.py [hunk 1]
```diff
@@ -1,47 +1,36 @@
...
-                          decay_after_epoch=5,
-                          lr_decay=0.5)
         params = [torch.nn.Parameter(torch.randn(2,3,4))]
-        optim.set_parameters(params)
-        optim.update(0, 10)
-        self.assertEquals(optim.optimizer.param_groups[0]['lr'], 0.5)
+        optimizer = Optimizer(torch.optim.Adam(params, lr=1), max_grad_norm=5)
+        scheduler = StepLR(optimizer.optimizer, 1, gamma=0.1)
+        optimizer.set_scheduler(scheduler)
+        optimizer.update(0, 10)
```
Comment (avinash2692) [inline tests/test_optim_optim.py]: Is this supposed to be  `optimizer.update(10, 0)` cause update takes input as `.update(loss, epoch)` ? but I guess it doesn't make a difference to the `lr`

File: seq2seq/optim/optim.py [hunk 2]
```diff
@@ -1,79 +1,51 @@
...
-            torch.nn.utils.clip_grad_norm(self.parameters, self.max_grad_norm)
+            params = itertools.chain.from_iterable([group['params'] for group in self.optimizer.param_groups])
+            torch.nn.utils.clip_grad_norm(params, self.max_grad_norm)
         self.optimizer.step()
 
     def update(self, loss, epoch):
-        """ Update the learning rate if the conditions are met. Override this method
-        to implement your own learning rate schedule.
-
+        """ Update the learning rate if the criteria of the scheduler are met.
```
Comment (avinash2692) [inline seq2seq/optim/optim.py]: do you need epoch for this method ?

Comment (kylegao91) [reply seq2seq/optim/optim.py]: No for now, but it could be useful if somebody override the method.

Comment (avinash2692) [reply seq2seq/optim/optim.py]: 👍
